File: string-paradigm-expression_2/src/test/java/org/jesperancinha/string/paradigm/original/expression/RegionSpellingCheckTest.java
package org.jesperancinha.string.paradigm.original.expression;

import static com.google.common.truth.Truth.assertThat;

import java.util.ArrayList;
import java.util.List;

import org.junit.Test;

public class RegionSpellingCheckTest {
	private static final String _1 = "1'";
	private static final String _2 = "2'";
	private static final String _3 = ",3";
	private static final String ALGARVE = "Algarve";
	private static final String RIBATEJO = "  Ribatejo";
	private static final String MINHO = "minho";

	@Test
	public void testCheckSpelling() {
		String[] regions = { ALGARVE, MINHO, RIBATEJO };
		String[] assignedNumbers = { _1, _2, _3 };
		List<RegionImpl> components = new ArrayList<>();
		for (String type : regions) {
			for (String size : assignedNumbers) {
				components.add(new RegionImpl(type, size));
			}
		}

		final RegionSpellingCheck check = new RegionSpellingCheck();
		for (RegionImpl component : components) {
			check.checkSpelling(component);
		}

		final List<RegionImpl> resultSpellingBeginCapitalRegions = check.getSpellingBeginCapitalRegions();
		assertThat(resultSpellingBeginCapitalRegions).hasSize(6);
		assertThat(resultSpellingBeginCapitalRegions.get(0).getRegionName()).startsWith(ALGARVE);
		assertThat(resultSpellingBeginCapitalRegions.get(1).getRegionName()).startsWith(ALGARVE);
		assertThat(resultSpellingBeginCapitalRegions.get(2).getRegionName()).startsWith(ALGARVE);
		assertThat(resultSpellingBeginCapitalRegions.get(3).getRegionName()).startsWith(RIBATEJO);
		assertThat(resultSpellingBeginCapitalRegions.get(4).getRegionName()).startsWith(RIBATEJO);
		assertThat(resultSpellingBeginCapitalRegions.get(5).getRegionName()).startsWith(RIBATEJO);

		final List<RegionImpl> resultSpellingDoubleCommaRegions = check.getSpellingDoubleCommaRegions();
		assertThat(resultSpellingDoubleCommaRegions).hasSize(3);
		for (RegionImpl region : resultSpellingDoubleCommaRegions) {
			assertThat(region.getAssignedNumber()).startsWith(_3);
		}

		final List<RegionImpl> resultSpellingWhiteSpacesRegions = check.getSpellingWhiteSpacesRegions();
		assertThat(resultSpellingWhiteSpacesRegions).hasSize(3);
		for (RegionImpl region : resultSpellingWhiteSpacesRegions) {
			assertThat(region.getRegionName()).startsWith(RIBATEJO);
		}
	}
}

File: string-paradigm-expression_2/src/test/kotlin/org/jesperancinha/string/paradigm/original/expression/RegionSpellingCheckTest.kt
package org.jesperancinha.string.paradigm.original.expression

import io.kotest.matchers.collections.shouldHaveSize
import io.kotest.matchers.string.shouldStartWith
import org.junit.jupiter.api.Test

class RegionSpellingCheckTest {
    @Test
    fun testCheckSpelling() {
        val regions = arrayOf(ALGARVE, MINHO, RIBATEJO)
        val assignedNumbers = arrayOf("1'", "2'", ",3")
        val components: MutableList<RegionImpl> = ArrayList()
        for (type in regions) {
            for (size in assignedNumbers) {
                components.add(RegionImpl(type, size))
            }
        }
        val check = RegionSpellingCheck()
        for (component in components) {
            check.checkSpelling(component)
        }
        val resultSpellingBeginCapitalRegions: List<RegionImpl> = check.spellingBeginCapitalRegions
        resultSpellingBeginCapitalRegions.shouldHaveSize(6)
        resultSpellingBeginCapitalRegions[0].regionName.shouldStartWith(ALGARVE)
        resultSpellingBeginCapitalRegions[1].regionName.shouldStartWith(ALGARVE)
        resultSpellingBeginCapitalRegions[2].regionName.shouldStartWith(ALGARVE)
        resultSpellingBeginCapitalRegions[3].regionName.shouldStartWith(RIBATEJO)
        resultSpellingBeginCapitalRegions[4].regionName.shouldStartWith(RIBATEJO)
        resultSpellingBeginCapitalRegions[5].regionName.shouldStartWith(RIBATEJO)
        val resultSpellingDoubleCommaRegions: List<RegionImpl> = check.spellingDoubleCommaRegions
        resultSpellingDoubleCommaRegions.shouldHaveSize(3)
        for (region in resultSpellingDoubleCommaRegions) {
            region.assignedNumber.shouldStartWith(",3")
        }
        val resultSpellingWhiteSpacesRegions: List<RegionImpl> = check.spellingWhiteSpacesRegions
        resultSpellingWhiteSpacesRegions.shouldHaveSize(3)
        for (region in resultSpellingWhiteSpacesRegions) {
            region.regionName.shouldStartWith(RIBATEJO)
        }
    }

    companion object {
        private const val ALGARVE = "Algarve"
        private const val RIBATEJO = "  Ribatejo"
        private const val MINHO = "minho"
    }
}